R/umap.R

```diff
@@ -105,6 +113,8 @@ step_umap <-
            metric = "euclidean",
            learn_rate = 1,
            epochs = NULL,
+           initial = "spectral",
+           target_weight = 0.5,
```

Comment:
I would default it to zero and, if it is not zero, check to make sure that the `outcome` is not NULL

Reply:
That would break backwards compatibility, computation-wise

in `main`, the following code uses `target_weight = 0.5` by default in ` uwot::umap()`.

```r
supervised <-
  recipe(Species ~ ., data = iris) %>%
  step_umap(all_predictors(), outcome = vars(Species), num_comp = 2) %>%
  prep()
```

With this proposed change, setting the `outcome` variable would no longer have an effect unless they also set the new `target_weight`.

Reply:
Ok, go ahead and default it to 1 / 2.  They ignore `target_weight` unless `y` is passed. This is a modified version of what is in `?umap`: 

``` r
library(uwot)
#> Loading required package: Matrix

ind <- c(1:10, 51:60, 101:110)

iris30 <- iris[ind,]

set.seed(1)
iris_umap_un_1 <-
  umap(
    iris30[, 1:4],
    n_neighbors = 5,
    learning_rate = 0.5,
    init = "random",
    n_epochs = 20,
    target_weight = 0.5
  )

set.seed(1)
iris_umap_un_2 <-
  umap(
    iris30[, 1:4],
    n_neighbors = 5,
    learning_rate = 0.5,
    init = "random",
    n_epochs = 20,
    target_weight = 0.0
  )

# just checking!
all.equal(iris_umap_un_1, iris_umap_un_2)
#> [1] TRUE

set.seed(1)
iris_umap_sup <-
  umap(
    iris30[, 1:4],
    n_neighbors = 5,
    learning_rate = 0.5,
    init = "random",
    n_epochs = 20,
    y = iris30$Species,
    target_weight = 0.5
  )

all.equal(iris_umap_un_1, iris_umap_sup)
#> [1] "Attributes: < Component \"scaled:center\": Mean relative difference: 0.6626731 >"
#> [2] "Mean relative difference: 0.3312773"
```

<sup>Created on 2024-01-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Note that in their man page, they pass the outcome in with the original data as well as with the `y` argument. That seems like a mistake. My example above avoids that and you get the same results in the first two cases regardless of the target weight.